src/libsystemd-network/sd-radv.c

```diff
@@ -112,33 +121,133 @@ _public_ sd_radv *sd_radv_unref(sd_radv *ra) {
                 sd_radv_prefix_unref(p);
         }
 
+        radv_reset(ra);
+
         sd_radv_detach_event(ra);
         return mfree(ra);
 }
 
+static int radv_send(sd_radv *ra, const struct in6_addr *dst,
+                     const uint32_t router_lifetime) {
+
+        return -ENOSYS;
+}
+
+static usec_t radv_compute_timeout(usec_t min, usec_t max) {
+        assert_return(min <= max, SD_RADV_DEFAULT_MIN_TIMEOUT_USEC);
+
+        return min + (random_u32() % (max - min));
+}
+
+static int radv_timeout(sd_event_source *s, uint64_t usec, void *userdata) {
+        int r;
+        sd_radv *ra = userdata;
+        usec_t min_timeout = SD_RADV_DEFAULT_MIN_TIMEOUT_USEC;
+        usec_t max_timeout = SD_RADV_DEFAULT_MAX_TIMEOUT_USEC;
+        usec_t time_now, timeout;
+        char time_string[FORMAT_TIMESPAN_MAX];
+
+        assert(s);
+        assert(ra);
+        assert(ra->event);
+
+        ra->timeout_event_source = sd_event_source_unref(ra->timeout_event_source);
+
+        r = sd_event_now(ra->event, clock_boottime_or_monotonic(), &time_now);
+        if (r < 0)
+                goto fail;
+
+        if (radv_send(ra, NULL, ra->lifetime) < 0)
+                log_radv("Unable to send Router Advertisement");
```

Comment:
If  `radv_send` returns, or may return, a meaningful error code, it might be worth to change this to use `%m`? `log_radv` would need to be changed, but such change would be also useful in other error messages.